website/integrations: Beszel

nicedevil007 opened the pull request:
## Details

Found a new tool that I tested and is able to use authentik for awesome login ;) Here is my guide.
For reference if you want to correct my english or just don't know what I wanted to describe without screenshots.
Here is the "default guide" found on the beszel homepage without the parts for authentik.

https://beszel.dev/guide/oauth

---

## Checklist

-   [X] Local tests pass (`ak test authentik/`)
-   [X] The code has been formatted (`make lint-fix`)

If an API change has been made

-   [ ] The API schema has been updated (`make gen-build`)

If changes to the frontend have been made

-   [ ] The code has been formatted (`make web`)

If applicable

-   [X] The documentation has been updated
-   [X] The documentation has been formatted (`make website`)

nicedevil007 committed:
initial release

nicedevil007 committed:
Ready for PR

nicedevil007 committed:
never changed this?

Used these commands after fresh git pull:

make lint-fix && npx prettier --write ./website/integrations/services/beszel/index.mdx && make website


Signed-off-by: NiceDevil <17103076+nicedevil007@users.noreply.github.com>

dominic-r reviewed (changes requested):
This service utilizes Pocketbase as its backend. I believe it would be more beneficial to focus this PR on configuring authentik alongside Pocketbase, as it would support more use cases.

nicedevil007 commented:
> This service utilizes Pocketbase as its backend. I believe it would be more beneficial to focus this PR on configuring authentik alongside Pocketbase, as it would support more use cases.

I agree with you but I don't know how I can do that without the real tests beeing done. As you might know, I'm not that devguy that deploy own applications and use backends like pocketbase on developing ;)
If there is a pocketbase guide anytime in the future we could change the one from Beszel?

nicedevil007 commented:
ok, so I can do another PR for pocketbase, you take a look at it, if this is not going the right direction we stay to you "only" Beszel docs ok?

nicedevil007 commented:
I can do it in parallel so maybe you will tell me "Nice...what are you trying to tell us with this guide..." and then this one is already close to be "merged" maybe :)

nicedevil007 commented:
> Yea, that's perfect

I ask here, because no new PR right now => do you have any information about the automatic/manual usercreation with pocketbase?

Is this a feature Beszel added to it or is it always the same for this backend?

nicedevil007 commented:
ok so I assume I can remove the part with the user creation from the guide at all, because we are using the docs in default state to get authentik up and running and not with user choosen details on each application ;)

nicedevil007 commented:
so... I guess I'm done with the code for PocketBase... do you have an example for an already existing integration that relies on any other service that I can use as a template for Beszel?

dominic-r reviewed (changes requested):
Hi @nicedevil007, Thanks for yet another conribution to authentik! Real quick, could you also apply the same suggestions from the Pocketbase PR? I'll review the remaining changes once that's done.

nicedevil007 committed:
reference to PocketBase

As @4d62 suggestet, here is the much simpler version as it uses 1:1 settings from PocketBase

nicedevil007 commented:
is here something missing from my side to get this PR merged?

tanberry commented:
> is here something missing from my side to get this PR merged?

Hi, as always, I am playing catchup, sorry! @dominic-r what else is needed, if anything, for this PR?

dominic-r commented on website/integrations/services/beszel/index.mdx:
```diff
@@ -0,0 +1,52 @@
+---
+
+# Integrate With Beszel
+
```
can you also rebase your pr with main and update this part? Support level should be moved here: https://github.com/goauthentik/authentik/blob/a07ce35985cdc12c57ca1ed43181ce1e1b743e98/website/integrations/template/service.md?plain=1#L4 and "Integrate with Beszel" should be removed

tanberry commented on website/integrations/services/beszel/index.mdx:
```diff
@@ -0,0 +1,52 @@
+
+## Preparation
+
+Since Beszel uses PocketBase as its backend, you only need to replace the PocketBase placeholders with the placeholders found on this page.
```
I think we need to more clearly explain the dependencies between PB and Beszel. And, even though it is not great to have to duplicate content, I don't think we should ask readers to have to go read other docs (the PB docs). I'll make a pass at a compromise idea.. we can see how it looks.

Question: does a user have to also configure PB as part of this Beszel setup? Or if they already set up PB before, do they not need to do anything?

nicedevil007 replied:
You install Beszel and it already comes with Pocketbase installed as its backend, no need for separate setup steps to get Pocketbase Running.

tanberry replied:
oh oh, so two possibilities:

- IF they already have integrated PB, they still have to also do these steps for Beszel (basically the same steps as for PB but diff FQDNs of course), 

- if they have not yet installed PB, and they want to install Beszel, then they DO NOT have to also install PB also. It will come with the PB integration.

Is this correct @nicedevil007 ? Thank you for your bravery and patience... :-)

nicedevil007 replied:
this is correct, I followed the comment here to split those up,. If I have to revert back to the only Beszel Docs without pocketbase I have to say sorry, we close this then and I don't know who I can trust on advices here anymore :(

https://github.com/goauthentik/authentik/pull/12905#pullrequestreview-2587822774

tanberry commented on website/integrations/services/beszel/index.mdx:
```diff
@@ -0,0 +1,52 @@
+
+## Preparation
+
+Since Beszel uses PocketBase as its backend, you only need to replace the PocketBase placeholders with the placeholders found on this page.
```
```suggestion
```

tanberry commented on website/integrations/services/beszel/index.mdx:
```diff
@@ -0,0 +1,52 @@
+:::note
+This documentation lists only the settings that you need to change from their default values. Be aware that any changes other than those explicitly mentioned in this guide could cause issues accessing your application.
+:::
+
```
```suggestion

## authentik configuration

Since Beszel uses PocketBase as its backend, when you integrate Beszel you can follow the exact same steps used to configure authentik as you did with PocketBase; you only need to replace the PocketBase placeholders with the placeholder FDQNs found above.

You can find the PocketBase documentation here: https://docs.goauthentik.io/integrations/services/pocketbase/.
```

dominic-r replied:
id use markdown to link this page and use references like ../path/to/file instead of direct links

tanberry commented on website/integrations/services/beszel/index.mdx:
```diff
@@ -0,0 +1,52 @@
+## Preparation
+
+Since Beszel uses PocketBase as its backend, you only need to replace the PocketBase placeholders with the placeholders found on this page.
+You can find the PocketBase documentation here: https://docs.goauthentik.io/integrations/services/pocketbase/.
```
```suggestion
```

tanberry reviewed (changes requested):
Thanks as always @nicedevil007 for your contributions! This PR is a bit complex, because of the dependency with PocketBase. Take a look at my suggestions and let us know what you think.

nicedevil007 committed:
index.mdx aktualisieren

Co-authored-by: Tana M Berry <tanamarieberry@yahoo.com>
Signed-off-by: NiceDevil <17103076+nicedevil007@users.noreply.github.com>

nicedevil007 committed:
index.mdx aktualisieren

Co-authored-by: Tana M Berry <tanamarieberry@yahoo.com>
Signed-off-by: NiceDevil <17103076+nicedevil007@users.noreply.github.com>

nicedevil007 committed:
index.mdx aktualisieren

Co-authored-by: Tana M Berry <tanamarieberry@yahoo.com>
Signed-off-by: NiceDevil <17103076+nicedevil007@users.noreply.github.com>

tanberry commented:
Hi @nicedevil007 if it is ok with you I am going to pull in  your PR and work on it locally, then push my changes as a PR into yours. ;-)

This is a complex one, and I want to make it clear that users don't have to install PB unless they want to.

nicedevil007 commented:
Sure, no Problem at all. 👍

Commit:
add full instructions

Commit:
add Beszel config

tanberry commented:
OK @nicedevil007 apologies for the complete reverse of earlier guidance from us, I went and added in all of the full steps for authentik configuration and Beszel configuration. It took me a bot to understand that Beszel and Pocketbook truly are independent enough that they need completely separate (even if 99% duplicated content) guides.

Please review especially the setps for the Beszel configuration. I used the steps from PB... please review URLs, etc.

I'm confident we will get this merged soon! Thank you again.

dominic-r commented on website/integrations/services/beszel/index.mdx:
```diff
@@ -0,0 +1,86 @@
+1. Log in to authentik as an admin, and open the authentik Admin interface.
+2. Navigate to **Applications** > **Applications** and click **Create with Provider** to create an application and provider pair.
+
+- **Application**: provide a descriptive name ()`Beszel`), a slug (`bezsel`), an optional group for the type of application, the policy engine mode, and optional UI settings.
```
It took you a bot and me... Anyways https://app.slack.com/archives/D08DJE7JYD7/p1741986549630009 
```suggestion
- **Application**: provide a descriptive name, a slug, an optional group for the type of application, the policy engine mode, and optional UI settings.
```

tanberry replied:
```suggestion
- **Application**: provide a descriptive name ()`Beszel`), a slug (`beszel`), an optional group for the type of application, the policy engine mode, and optional UI settings.
```

dominic-r commented on website/integrations/services/beszel/index.mdx:
```diff
@@ -0,0 +1,86 @@
+- **Choose a Provider type**: OAuth2/OpenID
+- **Configure the Provider**: provide a name (or accept the auto-provided name), the authorization flow to use for this provider, and any required configurations.
+    - Note the **Client ID**, **Client Secret**, and **slug** values because they will be required later.
+    - Set a `Strict` redirect URI to `https://pgadmin.company/oauth2/authorize`.
```
Same slack link +
```suggestion
    - Set a `Strict` redirect URI to `https://pgadmin.company/oauth2/authorize`.
```

I assume this is a typo?

dominic-r commented:
@tanberry Took you a bot and me :rofl: anyways, nits (internal only): https://app.slack.com/archives/D08DJE7JYD7/p1741986741812229

tanberry committed:
Update website/integrations/services/beszel/index.mdx

Signed-off-by: Tana M Berry <tanamarieberry@yahoo.com>

tanberry commented on website/integrations/services/beszel/index.mdx:
```diff
@@ -0,0 +1,86 @@
+1. Log in to authentik as an admin, and open the authentik Admin interface.
+2. Navigate to **Applications** > **Applications** and click **Create with Provider** to create an application and provider pair.
+
+- **Application**: provide a descriptive name ()`Beszel`), a slug (`beszel`), an optional group for the type of application, the policy engine mode, and optional UI settings.
```
```suggestion
- **Application**: provide a descriptive name (`Beszel`), a slug (`beszel`), an optional group for the type of application, the policy engine mode, and optional UI settings.
```

tanberry committed:
Update website/integrations/services/beszel/index.mdx

Signed-off-by: Tana M Berry <tanamarieberry@yahoo.com>

tanberry commented on website/integrations/services/beszel/index.mdx:
```diff
@@ -0,0 +1,86 @@
+- **Choose a Provider type**: OAuth2/OpenID
+- **Configure the Provider**: provide a name (or accept the auto-provided name), the authorization flow to use for this provider, and any required configurations.
+    - Note the **Client ID**, **Client Secret**, and **slug** values because they will be required later.
+    - Set a `Strict` redirect URI to `https://pgadmin.company/oauth2/authorize`.
```
```suggestion
    - Set a `Strict` redirect URI to `https://beszel.company/oauth2/authorize`.
```

dominic-r replied:
make sure you got the right redir url too and kbd/em please. see slack for corrected ver

dominic-r replied:
```suggestion
    - Set a `Strict` redirect URI to <kbd>https://<em>beszel.company</em>/oauth2/authorize</kbd>.
```

nicedevil007 replied:
correct one would be this `https://beszel.company/api/oauth2-redirect`
I hope my edit on the file was done correctly, never done a suggestion for edit of a file on github

dominic-r replied:
Thanks!

nicedevil007 committed:
correct redirect URI

Signed-off-by: NiceDevil <17103076+nicedevil007@users.noreply.github.com>

tanberry reviewed (approved):
Thank you so much @nicedevil007!!
